packages/protocol/ledger/storage.go

```diff
@@ -258,32 +258,58 @@ func (s *Storage) initConsumers(outputIDs utxo.OutputIDs, txID utxo.TransactionI
 // pruneTransaction removes a Transaction (and all of its dependencies) from the database.
 func (s *Storage) pruneTransaction(txID utxo.TransactionID, pruneFutureCone bool) {
```

Comment:
This entire method is currently only used upon orphanage, so it should be removed. We could although use it on pruning: as currently we store in the ledger storage every output that ever existed, regardless of the fact that it was already spent by a (now pruned) transaction.